main.tf

```diff
@@ -143,36 +143,55 @@ data "aws_ami" "user_ami" {
   }
 }
 
-resource "aws_launch_configuration" "container_instance" {
-  lifecycle {
-    create_before_destroy = true
+resource "aws_launch_template" "container_instance" {
+  block_device_mappings {
+    device_name = "/dev/sda1"
+
+    ebs {
+      volume_type = "${var.root_block_device_type}"
+      volume_size = "${var.root_block_device_size}"
+    }
   }
 
-  root_block_device {
-    volume_type = "${var.root_block_device_type}"
-    volume_size = "${var.root_block_device_size}"
+  credit_specification {
+    cpu_credits = "${var.unlimited_cpu ? "unlimited" : "standard"}"
   }
 
-  name_prefix          = "lc${title(var.environment)}ContainerInstance-"
-  iam_instance_profile = "${aws_iam_instance_profile.container_instance.name}"
+  disable_api_termination = false
+
+  name_prefix = "lt${title(var.environment)}ContainerInstance-"
+
+  iam_instance_profile {
+    name = "${aws_iam_instance_profile.container_instance.name}"
+  }
 
   # Using join() is a workaround for depending on conditional resources.
   # https://github.com/hashicorp/terraform/issues/2831#issuecomment-298751019
   image_id = "${var.lookup_latest_ami ? join("", data.aws_ami.ecs_ami.*.image_id) : join("", data.aws_ami.user_ami.*.image_id)}"
 
-  instance_type   = "${var.instance_type}"
-  key_name        = "${var.key_name}"
-  security_groups = ["${aws_security_group.container_instance.id}"]
-  user_data       = "${data.template_cloudinit_config.container_instance_cloud_config.rendered}"
+  instance_initiated_shutdown_behavior = "terminate"
+  instance_type                        = "${var.instance_type}"
+  key_name                             = "${var.key_name}"
+  vpc_security_group_ids               = ["${aws_security_group.container_instance.id}"]
+  user_data                            = "${base64encode(data.template_cloudinit_config.container_instance_cloud_config.rendered)}"
+
+  monitoring {
```

Comment:
The Terraform docs aren't very descriptive on what this actually means.

Reply:
It seems like this maps to https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch-new.html. This seems like a good variable to surface in the module and default to `false` (we'll probably set it to `true` most of the time).